### ISSUE TITLE
Date Filter : Next | Previous (Week/Month/etc) can't be set at 2 (0.21.0)

### ISSUE THREAD

**Issue by romone22** [metabase/metabase#3910]
Hi, 

It seams the offset for the NEXT / PREVIOUS Date filter is not right between 1 and 2. 

It's actually not possible to filter on the last 2 months for instance. Cf Screenshot 

<img width="1240" alt="screen shot 2016-12-09 at 9 56 23 am" src="https://cloud.githubusercontent.com/assets/5988677/21059191/20afe828-bdf6-11e6-815f-ed51c77473e7.png">
<img width="1235" alt="screen shot 2016-12-09 at 9 56 13 am" src="https://cloud.githubusercontent.com/assets/5988677/21059194/2100cbd0-bdf6-11e6-848e-1b8709192084.png">
<img width="1247" alt="screen shot 2016-12-09 at 9 56 01 am" src="https://cloud.githubusercontent.com/assets/5988677/21059192/20fda91e-bdf6-11e6-9f1a-194c2aa56275.png">
<img width="1230" alt="screen shot 2016-12-09 at 9 55 47 am" src="https://cloud.githubusercontent.com/assets/5988677/21059193/20fdffb8-bdf6-11e6-88fe-9f8fad70ed46.png">

**Comment by tlrobinson** [comment]
Possibly a duplicate of #3731

**Comment by camsaul** [comment]
Hi @romone22 , could you fill out our bug questionnaire so we can get a little more info in order to diagnose this issue?

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
- Your operating system: (e.x. OS X 10, Windows XP, etc)
- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
- Metabase version: (e.x. 0.19.3)
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Linux/Ubuntu 12)
- Metabase internal database: (e.x. H2 (default), MySQL, Postgres)

**Comment by salsakran** [comment]
@cam I've repro'd this on the sample dataset.

**Comment by camsaul** [comment]
Ok I see why this is happening. SQL `BETWEEN` is inclusive and the SQL looks like

```sql
SELECT ...
WHERE created_at BETWEEN [3 months ago] AND [1 month ago]
...
```

The SQL should actually be `BETWEEN [2 months ago] AND [1 month ago]`.

**Comment by camsaul** [comment]
Fixed by #4101
